runhouse/resources/hardware/cluster.py

```diff
@@ -1959,3 +1967,202 @@ def _folder_mv(
 
     def _folder_exists(self, path: Union[str, Path]):
         return self.client.folder_exists(path=path)
+
+    ###################################
+    # Cluster list helping methods
+    ###################################
+
+    @staticmethod
+    def _parse_time_duration(duration: str):
```

Comment:
some of these feel more like pure "utils", and less specific to the cluster class specifically. thoughts on moving these into the `utils.py` instead?